Mage.Sets/src/mage/cards/h/HenryWuInGenGeneticist.java

```diff
@@ -0,0 +1,103 @@
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class HenryWuInGenGeneticist extends CardImpl {
+
+    public static final FilterPermanent filterYourHumans = new FilterCreaturePermanent("Human creatures you control");
+
+    static {
+        filterYourHumans.add(TargetController.YOU.getControllerPredicate());
+        filterYourHumans.add(SubType.HUMAN.getPredicate());
+    }
+
+    public static final FilterPermanent filterNonHumans = new FilterCreaturePermanent("non-Human creature");
+
+    static {
+        filterNonHumans.add(Predicates.not(SubType.HUMAN.getPredicate()));
+    }
+
+    public HenryWuInGenGeneticist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{B}{G}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SCIENTIST);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(4);
+
+        // Henry Wu, InGen Geneticist and other Human creatures you control have exploit.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityControlledEffect(
+                new ExploitAbility(), Duration.WhileOnBattlefield, filterYourHumans)));
+
+        // Whenever a creature you control exploits a non-Human creature, draw a card. If the exploited creature had power 3 or greater, create a Treasure token.
+        this.addAbility(new HenryWuInGenGeneticistTriggeredAbility(filterNonHumans));
+    }
+
+    private HenryWuInGenGeneticist(final HenryWuInGenGeneticist card) {
+        super(card);
+    }
+
+    @Override
+    public HenryWuInGenGeneticist copy() {
+        return new HenryWuInGenGeneticist(this);
+    }
+}
+
+// Based on ColonelAutumnTriggeredAbility
+class HenryWuInGenGeneticistTriggeredAbility extends TriggeredAbilityImpl {
+
+    private final FilterPermanent filter;
+
+    HenryWuInGenGeneticistTriggeredAbility(FilterPermanent filter) {
+        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(1));
+//        this.addEffect(new ConditionalOneShotEffect());
+        this.filter = filter;
+        setTriggerPhrase("Whenever a creature you control exploits a " + this.filter.getMessage());
+    }
+
+    private HenryWuInGenGeneticistTriggeredAbility(final HenryWuInGenGeneticistTriggeredAbility ability) {
+        super(ability);
+        this.filter = ability.filter;
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.EXPLOITED_CREATURE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent exploiter = game.getPermanentOrLKIBattlefield(event.getSourceId());
+        Permanent exploited = game.getPermanentOrLKIBattlefield(event.getTargetId());
+
+        return exploiter != null && exploited != null
+                && exploiter.isCreature(game)
```

Comment:
No need to check for isCreature if the filter also checks it.
You could also add the following predicate to it and avoid `isControlledBy`: `TargetController.YOU.getControllerPredicate();`, but that's more code style than actually incorrect.

About the filter, use the 4-argument `match` instead like this:
`filter.match(exploited, getControllerId(), this, game)`

Some `Predicates` are not checked by the 2-argument ones (the controller predicate above for instance), and there are a few others quirks like not checking if the permanent is phased in (doesn't matter for Henry).